receiver/splunkhecrawreceiver/receiver.go

```diff
@@ -0,0 +1,263 @@
+	mx := mux.NewRouter()
+	mx.NewRoute().HandlerFunc(r.handleReq)
+
+	r.server = r.config.HTTPServerSettings.ToServer(mx)
+
+	// TODO: Evaluate what properties should be configurable, for now
+	//		set some hard-coded values.
+	r.server.ReadHeaderTimeout = defaultServerTimeout
+	r.server.WriteTimeout = defaultServerTimeout
+	r.resourceCustomizer = r.createResourceCustomizer()
+
+	go func() {
+		if errHTTP := r.server.Serve(ln); errHTTP != http.ErrServerClosed {
+			host.ReportFatalError(errHTTP)
+		}
+	}()
+
+	return err
+}
+
+// Shutdown tells the receiver that should stop reception,
+// giving it a chance to perform any necessary clean-up.
+func (r *splunkReceiver) Shutdown(context.Context) error {
+	return r.server.Close()
+}
+
+func (r *splunkReceiver) handleReq(resp http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if r.logsConsumer == nil {
+		ctx = r.obsrecv.StartMetricsOp(ctx)
+	}
+	reqPath := req.URL.Path
+	if !r.config.pathGlob.Match(reqPath) {
+		r.failRequest(ctx, resp, http.StatusNotFound, notFoundRespBody, nil)
+		return
+	}
+
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, nil)
+		return
+	}
+
+	encoding := req.Header.Get(httpContentEncodingHeader)
+	if encoding != "" && encoding != gzipEncoding {
+		r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidEncodingRespBody, nil)
+		return
+	}
+
+	bodyReader := req.Body
+	if encoding == gzipEncoding {
+		var err error
+		bodyReader, err = gzip.NewReader(bodyReader)
+		if err != nil {
+			r.failRequest(ctx, resp, http.StatusBadRequest, errGzipReaderRespBody, err)
+			return
+		}
+	}
+
+	if req.ContentLength == 0 {
+		resp.Write(okRespBody)
+		return
+	}
+
+	sc := bufio.NewScanner(bodyReader)
+
+	ld := pdata.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	r.resourceCustomizer(req, rl.Resource())
+	ill := rl.InstrumentationLibraryLogs().AppendEmpty()
+
+	for sc.Scan() {
+		logRecord := ill.Logs().AppendEmpty()
+		logLine := sc.Text()
+		logRecord.Body().SetStringVal(logLine)
+	}
+	decodeErr := r.logsConsumer.ConsumeLogs(ctx, ld)
+
+	if decodeErr != nil {
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, decodeErr)
+	} else {
+		resp.WriteHeader(http.StatusAccepted)
+		resp.Write(okRespBody)
+	}
+}
+
+func (r *splunkReceiver) failRequest(
+	ctx context.Context,
+	resp http.ResponseWriter,
+	httpStatusCode int,
+	jsonResponse []byte,
+	err error,
+) {
+	resp.WriteHeader(httpStatusCode)
+	if len(jsonResponse) > 0 {
+		// The response needs to be written as a JSON string.
+		_, writeErr := resp.Write(jsonResponse)
+		if writeErr != nil {
+			r.logger.Warn("Error writing HTTP response message", zap.Error(writeErr))
+		}
+	}
+
+	msg := string(jsonResponse)
+
+	reqSpan := trace.FromContext(ctx)
+	reqSpan.AddAttributes(
+		trace.Int64Attribute(conventions.AttributeHTTPStatusCode, int64(httpStatusCode)),
+		trace.StringAttribute(conventions.AttributeHTTPStatusText, msg))
```

Comment:
I don't think it's conventional to add a json response body as status text

Reply:
Hmm - now I'm confused. The import is clearly from the collector core, yet I don't see this status text attribute even defined in the spec

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes

I suspect it might be some drift left over from an opencensus conversion but shouldn't actually exist. Would it be ok to leave it out from this code, to match the opentelemetry spec?

Reply:
Removed.